clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1182,6 +1182,9 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     const llvm::Triple &DeviceTriple = TI.getTriple();
     if (DeviceTriple.isNVPTX())
       Builder.defineMacro("__SYCL_NVPTX__", "1");
+
+    if (DeviceTriple.isAMDGCN())
+      Builder.defineMacro("__SYCL_AMDGCN__", "1");
```

Comment:
(Since `__SYCL_NVPTX__` is being removed in https://github.com/intel/llvm/pull/3977)
```suggestion
```